strings/base_array.h

```diff
@@ -345,6 +345,26 @@ WINRT_EXPORT namespace winrt
                 this->m_size = size;
             }
         }
+
+        std::pair<uint32_t, impl::arg_out<T>> detach_abi() noexcept
+        {
+#ifdef _MSC_VER
+            // https://github.com/microsoft/cppwinrt/pull/1165
+            std::pair<uint32_t, impl::arg_out<T>> result;
+            memset(&result, 0, sizeof(result));
+            result.first = this->size();
+            result.second = *reinterpret_cast<impl::arg_out<T>*>(this);
+            memset(this, 0, sizeof(com_array<T>));
+#else
+            std::pair<uint32_t, impl::arg_out<T>> result(this->size(), *reinterpret_cast<impl::arg_out<T>*>(this));
```

Comment:
Doesn't this reintroduce the uninitialized padding data for non-VC compilers, per the comments in the PR linked above? What compiler warning does this resolve?

Reply:
The GCC warning is `'void* memset(void*, int, size_t)' clearing an object of type 'struct std::pair<unsigned int, float*>' with no trivial copy-assignment; use assignment or value-initialization instead [-Wclass-memaccess]`.

This change does leave the padding uninitialized, but as said before it is "almost never going to be an issue for this function" and I am uncomfortable with the attempt to fix it potentially invoking UB. (Though I am no language lawyer so I cannot say for sure whether there is UB.)